yt/visualization/accumulators.py

```diff
@@ -0,0 +1,348 @@
+
+    fieldVals : YTArray 
+        An array containing the values of the scalar field to be
+        integrated at the location of the starting point of each
+        path segment as well as the endpoint for the last segment
+
+    Returns
+    -------
+    accum : YTArray 
+        The cumulative value of the field integral at each path
+        segment 
+    """
+    # https://en.wikipedia.org/wiki/Line_integral
+    # np.linalg.norm returns a ndarray, so when multiplied by field_vals, which
+    # is a YTArray, this leads to incorrect units. There does not appear to be
+    # a unyt implementation of norm, as far as I'm aware, so units will be
+    # handled manually for the time being
+    accum = np.cumsum(field_vals[:-1].d * np.linalg.norm(p[1:].d - p[:-1].d, axis=1))
+    accum = YTArray(accum, field_vals.units * p.units)
+    return accum
+
+
+def get_row_major_index(ncells, cell_ind):
+    """
+    Converts the cell indices (i_1, i_2, ..., i_N) to the corresponding
+    row-major index I.
+
+    The field data for each cell in the node are stored in an N dimensional
+    array with shape (n_1, n_2, ..., n_N), where n_i is the number of cells along
+    dimension i. The N dimensional index of the current cell under consideration,
+    `cell_ind` is (i_1, i_2, ..., i_N). However, accessing the value of a single
+    element in a multi-dimensional array (unflattened data) with another array
+    (cell_ind) is problematic.
+
+    As such, we flatten the data, which puts the data into row-major format.
+    The current cell under consideration can then be accessed by converting
+    cell_ind to the corresponding row-major index I.
+
+    The location I in the row-major array of the cell given by (1_1, i_2, ..., i_N)
+    is:
+
+        I = i_N + n_N * (i_{N-1} + n_{N-1} * (...(i_2 + n_2 * i)...)
+
+    which in three dimensions is:
+
+        I = k + n_z * (j + n_y * i)
+
+    Parameters
+    ----------
+    ncells : tuple
+        Contains the number of cells along each dimension.
+
+    cell_ind : tuple
+        Contains the integer values of the index of the current cell in each
+        dimension (i.e., this is (i_1, i_2, ..., i_N)).
+
+    Returns
+    -------
+    I : int
+        The row-major index corresponding to cell_ind.
+    """
+    I = cell_ind[0]
+    for idx in range(1, len(ncells)):
+        I = cell_ind[idx] + ncells[idx] * I
+    return I
+
+
+class Accumulators:
+    r"""
+    Container for creating and storing the path integrals of various
+    fields along various paths.
+
+    The class takes in a list of user-defined paths and a dataset. From
+    these, the user can compute the integral of any field in the dataset
+    along the given paths. The results of these path integrals are
+    stored in a cumulative fashion, which means that, once computed, the
+    accumulated value of the field is stored at each point along the
+    path, allowing the user to query this information.
+
+    Attributes
```

Comment:
I think you can get rid of these lines, since you don't fill them out, and generally we don't include stuff in the main docstring for these attributes anyway.